kubernetes/skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldFileService.kt

```diff
@@ -38,19 +38,37 @@ class SkaffoldFileService {
      * Checks if a given file is a valid Skaffold configuration file based on type and API version.
      */
     fun isSkaffoldFile(file: VirtualFile): Boolean {
-        with(file) {
-            if (!isDirectory && fileType is YAMLFileType && isValid) {
-                try {
-                    val skaffoldYaml = SkaffoldYamlConfiguration(this)
-                    return skaffoldYaml.apiVersion?.startsWith(SKAFFOLD_API_HEADER) == true
-                } catch (ex: Exception) {
-                    // We don't care about I/O or scan exceptions here since we only need to know if
-                    // the YAML file was in the proper format.
-                    return false
+        return try {
+            isValidYamlFile(file) && isValidSkaffoldFile(SkaffoldYamlConfiguration(file))
+        } catch (ex: Exception) {
+            // We don't care about I/O or scan exceptions here since we only need to know if
+            // the YAML file was in the proper format.
+            false
+        }
+    }
+
+    /**
+     * Returns the version of Skaffold as defined in skaffold.yaml, or null if it can't be read.
+     *
+     * The version is represented as skaffold/[version] in the yaml.
+     */
+    fun getSkaffoldVersion(file: VirtualFile): String? {
```

Comment:
Looking at this method and its using YAML data, how about actually moving it into `SkaffoldYamlConfiguration` itself?

Reply:
At least the version extraction part

Reply:
agreed. ill do that

Reply:
done